content/chainguard/chainguard-images/videos/repro.md

```diff
@@ -0,0 +1,236 @@
+---
+title: "Reproducibility and Chainguard Images"
+linktitle: "Reproducibility and Chainguard Images"
+lead: ""
+description: "This video explains the importance of reproducibility and how to reproduce any
+Chainguard image from an attestation."
+type: "article"
+date: 2024-05-20T12:21:01+00:00
+lastmod: 2024-05-20T12:21:01+00:00
+draft: false
+images: []
+menu:
+  docs:
+    parent: "chainguard-images"
+weight: 070
+toc: true
+---
+
+{{< youtube 0Qn2J89UEvI >}}
+
+## Clarification
+
+In this video we mention needing to keep copies of old APKs in order to be able to recreate images.
+This wasn't fully accuate -- in fact we do keep all our previously issued APKs, so you can build
+images from months (and in the future, years) ago without issue. We currently retain all of these
+package versions indefinitely (only servicing latest), but in the future we may age things out just
+to manage the size of the index.
+
+## Tools used in this video
+
+* [cosign](https://github.com/sigstore/cosign)
+* [apko](https://github.com/chainguard-dev/apko)
+* [diffoci](https://github.com/reproducible-containers/diffoci)
+
+## Commands
+
+Retrieving the build configuration for the latest version of nginx:
+
+```
+cosign verify-attestation \
+  --type https://apko.dev/image-configuration \
+  --certificate-oidc-issuer https://token.actions.githubusercontent.com \
+  --certificate-identity https://github.com/chainguard-images/images/.github/workflows/release.yaml@refs/heads/main \
+  cgr.dev/chainguard/nginx:latest | jq -r .payload | base64 -d | jq .predicate > latest.apko.json
+```
+
+Building the image:
+
+```
+apk publish latest.apko.json ttl.sh/nginx-repro
```

Comment:
This should be `apko`, not `apk`, right?

Reply:
yeah, I wonder how that happened, it should have been a cut and paste!